### PR TITLE
Fancy Auctions Browser: The Sequel

### PR DESCRIPTION
**Pull request by viciscat**
_Uh hello? Welcome to the new and improved "Fancy Auctions Browser" PR. Now, I want you to forget everything you've heard or seen in the previous PR._
This is a rewrite of my old PR to make it actually work and make it less stupid code wise (hopefully).

<details>
  <summary>Excessively large images from the previous PR. Same thing, give or take a few differences.</summary>

image

image

image

image

</details>

Responding to me in the previous PR: yes you missed edge cases you IDIOT. There might still be some edge cases.

### CODE REVIEW

File: src/main/resources/skyblocker.mixins.json [hunk 1]
```diff
@@ -50,5 +50,8 @@
   ],
   "injectors": {
     "defaultRequire": 1
-  }
+  },
+  "mixins": [
+    "accessor.SlotAccessor"
+  ]
```
Comment (kevinthegreat1) [inline src/main/resources/skyblocker.mixins.json:56]: You probably generated this accessor, which is fine, but just remember to put it in `client` with the other mixins

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
Code looks pretty good.

File: src/main/java/de/hysky/skyblocker/mixin/accessor/SlotAccessor.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Mutable;
+import org.spongepowered.asm.mixin.gen.Accessor;
+
+@Mixin(Slot.class)
+public interface SlotAccessor {
+
+    @Mutable
+    @Accessor("x")
+    void setX(int x);
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/mixin/accessor/SlotAccessor.java:13]: mm you could add a space here

File: src/main/resources/skyblocker.mixins.json [hunk 3]
```diff
@@ -48,9 +48,11 @@
     "accessor.ScreenAccessor",
     "accessor.SkullBlockEntityAccessor",
     "accessor.WorldRendererAccessor",
-    "discordipc.ConnectionMixin"
+    "discordipc.ConnectionMixin",
+    "accessor.SlotAccessor"
```
Comment (AzureAaron) [inline src/main/resources/skyblocker.mixins.json:52]: Move this to the rest of them so its alphabetical

File: src/main/resources/skyblocker.mixins.json [hunk 4]
```diff
@@ -48,9 +48,11 @@
...
   ],
   "injectors": {
     "defaultRequire": 1
-  }
+  },
+  "mixins": []
```
Comment (AzureAaron) [inline src/main/resources/skyblocker.mixins.json:57]: remove this

**Review by AzureAaron — CHANGES_REQUESTED** [review]

**Review by kevinthegreat1 — APPROVED** [review]
Works well. I fixed some small things. Item prices should be represented as longs since they can go over the integer limit. I also used `ItemUtils.getNbtTooltips` instead of `stack.getTooltip(client.player, TooltipContext.BASIC)`, which saves a bit of processing.

There is stil a bug that when a slot changes from and item to empty, it doesn't get updated, resulting in the cheap highlight still showing. If this can't be properly fixed, we can just not render the highlight if the slot is empty.

**Pull request merged**